java/UltraHdrApp.java

```diff
@@ -0,0 +1,705 @@
+/*
+ * Copyright 2024 The Android Open Source Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import static com.google.media.codecs.ultrahdr.UltraHDRCommon.*;
+import static com.google.media.codecs.ultrahdr.UltraHDREncoder.UHDR_USAGE_REALTIME;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.BufferedReader;
+import java.io.FileReader;
+import java.io.BufferedWriter;
+import java.io.FileWriter;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.nio.ByteBuffer;
+import java.nio.ByteOrder;
+
+import com.google.media.codecs.ultrahdr.UltraHDRDecoder;
+import com.google.media.codecs.ultrahdr.UltraHDREncoder;
+import com.google.media.codecs.ultrahdr.UltraHDRDecoder.GainMapMetadata;
+import com.google.media.codecs.ultrahdr.UltraHDRDecoder.RawImage;
+
+/**
+ * Ultra HDR Encoding/Decoding Demo Application
+ */
+public class UltraHdrApp {
```

Comment:
I assume this file can be directly imported by Java users, so maybe "UltraHdrApi" is better name as it aligns with what we currently have for C code ultrahdr_api.h. Other than this this change looks really nice! Thank you for making this happen!

Reply:
This is a sample application and this is importing uhdr encoder and uhdr decoder APIs via the import statements.
import com.google.media.codecs.ultrahdr.UltraHDRDecoder
import com.google.media.codecs.ultrahdr.UltraHDREncoder
They both combine together to form ultrahdr_api.h
Do you want us to rename those classes?

Reply:
OK do we have a "ultrahdr_api.h" - like java version file so that app developer can directly import that one, or it will be in the follow-up CL, or does app developer need to import UltraHDRDecoder and UltraHDREncoder separately?

Reply:
App developer will need to import UltraHDRDecoder and UltraHDREncoder.

Reply:
OK I see Thank you for clarification!